### PR TITLE
Fixes #5645: Fixes session undefined error when user has a default taxon...

### PR DESCRIPTION
**Pull request by ehelms**
...omy set.

### CODE REVIEW

**Comment by dLobatog** [comment]
It's failing because `options[:session]` is not set in an users controller test, should be easy to fix. Other than that :+1:

File: app/controllers/concerns/foreman/controller/authentication.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def sso_authentication
 
   def set_current_user(user)
     User.current = user
-    Foreman::Controller::UsersMixin.set_current_taxonomies(user)
+    Foreman::Controller::UsersMixin.set_current_taxonomies(user, {:session => session})
```
Comment (domcleal) [inline app/controllers/concerns/foreman/controller/authentication.rb]: As per the comment below, whether to update the session or not should perhaps depend upon `api_request?`.

Comment (ehelms) [inline app/controllers/concerns/foreman/controller/authentication.rb]: Are you saying you'd like me to bake that into this fix?

Comment (domcleal) [inline app/controllers/concerns/foreman/controller/authentication.rb]: Yeah, if not we'll have to get it addressed before including this IMO.

Comment (ehelms) [inline app/controllers/concerns/foreman/controller/authentication.rb]: Should it depend upon whether or not this is an API request? What if I am accessing the API as a session user?

**Comment by lzap** [comment]
How do I reproduce this one?

**Comment by lzap** [comment]
Unable to repro on a clean nightly Foreman, do I need Foretello for this?

File: app/controllers/concerns/foreman/controller/authentication.rb (hunk 1 above)
Comment (domcleal) [inline app/controllers/concerns/foreman/controller/authentication.rb]: I'm not sure the current taxonomy should even be set when using the API, as there's no notion of context supported there already.  My main concern though is that authenticating against the API shouldn't modify the session.

If you're accessing the API as a session user, I don't see that you'll hit this code, as you'd take the fast return out of #authenticate.  This code I think is only run during authentication (e.g. basic auth).

I think this line should move into the block below (`unless api_request?`), next to `session[:user]`.  Does that seem correct?

**Comment by ehelms** [comment]
@lzap I added steps to the issue on how I produced it (http://projects.theforeman.org/issues/5645)

**Comment by ehelms** [comment]
@domcleal updated

**Comment by lzap** [comment]
Confirmed, the patch works.

**Comment by domcleal** [comment]
Merged as e44762c for Foreman 1.5.1, thanks @ehelms!

**Pull request closed without merging**